In [ ]:
#!pip install yfinance

In [109]:
import yfinance as yf
import pandas as pd
import numpy as np
from datetime import datetime
import time

### Checking YFin API

In [ ]:
tickers_list = ["AAPL"]
tickers_data= {}

for ticker in tickers_list:
    ticker_object = yf.Ticker(ticker)

    #convert info() output from dictionary to dataframe
    temp = pd.DataFrame.from_dict(ticker_object.info, orient="index")
    temp.reset_index(inplace=True)
    temp.columns = ["Attribute", "Recent"]
    tickers_data[ticker] = temp

# tickers_data["Attribute"].unique()
ticker_object.info

{'address1': 'One Apple Park Way',
 'city': 'Cupertino',
 'state': 'CA',
 'zip': '95014',
 'country': 'United States',
 'phone': '(408) 996-1010',
 'website': 'https://www.apple.com',
 'industry': 'Consumer Electronics',
 'industryKey': 'consumer-electronics',
 'industryDisp': 'Consumer Electronics',
 'sector': 'Technology',
 'sectorKey': 'technology',
 'sectorDisp': 'Technology',
 'longBusinessSummary': 'Apple Inc. designs, manufactures, and markets smartphones, personal computers, tablets, wearables, and accessories worldwide. The company offers iPhone, a line of smartphones; Mac, a line of personal computers; iPad, a line of multi-purpose tablets; and wearables, home, and accessories comprising AirPods, Apple TV, Apple Watch, Beats products, and HomePod. It also provides AppleCare support and cloud services; and operates various platforms, including the App Store that allow customers to discover and download applications and digital content, such as books, music, video, games, and p

### Read CSV from BlackRock and CompaniesMarketCap

In [ ]:
# BlackRock file
r_df = pd.read_csv('russell3000holdings.csv')

# List of tickers from BlackRock file
r3000_tickers = r_df['Ticker'].tolist()

# CompaniesMarketCap file
emp_df = pd.read_csv('r3000_emp.csv')
emp_df

Rank                                  Name     Symbol  employees_count  \
0         1                               Walmart        WMT          2100000   
1         2                                Amazon       AMZN          1551000   
2         3  Foxconn (Hon Hai Precision Industry)    2317.TW           826608   
3         4                             Accenture        ACN           774000   
4         5                            Volkswagen    VOW3.DE           656134   
...     ...                                   ...        ...              ...   
9995   9996                    Dubai Islamic Bank     DIB.AE                0   
9996   9997                       Amanat Holdings  AMANAT.AE                0   
9997   9998                           Bank Muscat    BKMB.OM                0   
9998   9999                Omantel (Oman Telecom)    OTEL.OM                0   
9999  10000                OQ Gas Network Company    OQGN.OM                0   

      price (USD)               country  
0       89.760000         United States  
1      196.010000         United States  
2        4.627400                Taiwan  
3      316.330000               Ireland  
4      102.817000               Germany  
...           ...                   ...  
9995     1.954860  United Arab Emirates  
9996     0.313104  United Arab Emirates  
9997     0.657143                  Oman  
9998     2.207790                  Oman  
9999     0.345455                  Oman  

[10000 rows x 6 columns]

In [485]:
emp_df = emp_df[['Symbol', 'employees_count']]
emp_df.columns = ['symbol', 'emp_count']
emp_df

symbol  emp_count
0           WMT    2100000
1          AMZN    1551000
2       2317.TW     826608
3           ACN     774000
4       VOW3.DE     656134
...         ...        ...
9995     DIB.AE          0
9996  AMANAT.AE          0
9997    BKMB.OM          0
9998    OTEL.OM          0
9999    OQGN.OM          0

[10000 rows x 2 columns]

In [ ]:
r3000_df = r_df.iloc[:,[0,1,2,10]]
r3000_df.columns = ['symbol', 'name', 'sector', 'exchange']
r3000_df

symbol                                name                   sector  \
0      AAPL                           APPLE INC   Information Technology   
1      MSFT                      MICROSOFT CORP   Information Technology   
2      NVDA                         NVIDIA CORP   Information Technology   
3      AMZN                      AMAZON COM INC   Consumer Discretionary   
4      META          META PLATFORMS INC CLASS A            Communication   
...     ...                                 ...                      ...   
2671      -      OMNIAB INC $12.50 VESTING Prvt              Health Care   
2672  WLLBW  CHORD ENERGY EQUITY WARRANTS SERIE                   Energy   
2673  RTYM5       RUSSELL 2000 EMINI CME JUN 25  Cash and/or Derivatives   
2674   ESM5                 S&P500 EMINI JUN 25  Cash and/or Derivatives   
2675                                        NaN                      NaN   

                              exchange  
0                               NASDAQ  
1                               NASDAQ  
2                               NASDAQ  
3                               NASDAQ  
4                               NASDAQ  
...                                ...  
2671         NO MARKET (E.G. UNLISTED)  
2672  Non-Nms Quotation Service (Nnqs)  
2673       Chicago Mercantile Exchange  
2674          Index And Options Market  
2675                               NaN  

[2676 rows x 4 columns]

### Pull YFinance Data

In [501]:
# Exploring Data
start_date = "2025-01-20"
end_date = "2025-03-22"

companies = ['AMZN','GOOG','WMT','TSLA','META','METCB'] 
tickers = yf.Tickers(companies)
tickers_hist = tickers.history(period='max',start=start_date, end=end_date,interval='1d', auto_adjust=False)
tickers_hist.stack(level=1).rename_axis(['Date', 'Ticker']).reset_index(level=1)

Could not get exchangeTimezoneName for ticker 'METCB' reason: 'chart'
[*********************100%***********************]  6 of 6 completed

1 Failed download:
['METCB']: YFTzMissingError('possibly delisted; no timezone found')
/var/folders/4d/xk8pcdy56v12bg2qx66kdblc0000gn/T/ipykernel_91043/3841367413.py:8: FutureWarning: The previous implementation of stack is deprecated and will be removed in a future version of pandas. See the What's New notes for pandas 2.1.0 for details. Specify future_stack=True to adopt the new implementation and silence this warning.
  tickers_hist.stack(level=1).rename_axis(['Date', 'Ticker']).reset_index(level=1)


Price      Ticker   Adj Close       Close  Dividends        High         Low  \
Date                                                                           
2025-01-21   AMZN  230.710007  230.710007      0.000  231.779999  226.940002   
2025-01-21   GOOG  199.402832  199.630005      0.000  203.839996  199.440002   
2025-01-21   META  615.912048  616.460022      0.000  621.500000  609.010010   
2025-01-21   TSLA  424.070007  424.070007      0.000  433.200012  406.309998   
2025-01-21    WMT   92.825096   93.080002      0.000   94.239998   92.309998   
...           ...         ...         ...        ...         ...         ...   
2025-03-21   AMZN  196.210007  196.210007      0.000  196.990005  192.520004   
2025-03-21   GOOG  166.250000  166.250000      0.000  166.470001  163.029999   
2025-03-21   META  596.250000  596.250000      0.000  597.539978  580.950012   
2025-03-21   TSLA  248.710007  248.710007      0.000  249.520004  234.550003   
2025-03-21    WMT   85.980003   85.980003      0.235   86.230003   84.779999   

Price             Open  Stock Splits       Volume  
Date                                               
2025-01-21  228.899994           0.0   39951500.0  
2025-01-21  200.509995           0.0   19005200.0  
2025-01-21  617.830017           0.0   11674000.0  
2025-01-21  432.640015           0.0   87320900.0  
2025-01-21   92.480003           0.0   23247500.0  
...                ...           ...          ...  
2025-03-21  192.899994           0.0   60056900.0  
2025-03-21  163.380005           0.0   29882100.0  
2025-03-21  583.419983           0.0   25015900.0  
2025-03-21  234.990005           0.0  132728700.0  
2025-03-21   85.279999           0.0   26797200.0  

[215 rows x 9 columns]

In [ ]:
# Transposing data and pulling subset
companies = ['AMZN','GOOG','WMT','TSLA','META','METCB']
tickers = yf.Tickers(companies)
tickers_hist = tickers.history(period='max',start=start_date, end=end_date,interval='1d', auto_adjust=False)
tickers_hist = tickers_hist.T.reset_index()
tickers_hist[~tickers_hist.iloc[:, 2:].isnull().all(axis=1)]

Could not get exchangeTimezoneName for ticker 'METCB' reason: 'chart'
[*********************100%***********************]  6 of 6 completed

1 Failed download:
['METCB']: YFTzMissingError('possibly delisted; no timezone found')


Date         Price Ticker  2023-01-03 00:00:00  2023-01-04 00:00:00  \
0        Adj Close   AMZN         8.582000e+01         8.514000e+01   
1        Adj Close   GOOG         8.927714e+01         8.829181e+01   
2        Adj Close   META         1.241549e+02         1.267725e+02   
4        Adj Close   TSLA         1.081000e+02         1.136400e+02   
5        Adj Close    WMT         4.646152e+01         4.651329e+01   
6            Close   AMZN         8.582000e+01         8.514000e+01   
7            Close   GOOG         8.970000e+01         8.871000e+01   
8            Close   META         1.247400e+02         1.273700e+02   
10           Close   TSLA         1.081000e+02         1.136400e+02   
11           Close    WMT         4.786667e+01         4.792000e+01   
12       Dividends   AMZN         0.000000e+00         0.000000e+00   
13       Dividends   GOOG         0.000000e+00         0.000000e+00   
14       Dividends   META         0.000000e+00         0.000000e+00   
15       Dividends   TSLA         0.000000e+00         0.000000e+00   
16       Dividends    WMT         0.000000e+00         0.000000e+00   
17            High   AMZN         8.696000e+01         8.698000e+01   
18            High   GOOG         9.155000e+01         9.124000e+01   
19            High   META         1.263700e+02         1.290500e+02   
21            High   TSLA         1.188000e+02         1.145900e+02   
22            High    WMT         4.790333e+01         4.801667e+01   
23             Low   AMZN         8.421000e+01         8.336000e+01   
24             Low   GOOG         8.902000e+01         8.780000e+01   
25             Low   META         1.222800e+02         1.258500e+02   
27             Low   TSLA         1.046400e+02         1.075200e+02   
28             Low    WMT         4.736333e+01         4.735667e+01   
29            Open   AMZN         8.546000e+01         8.655000e+01   
30            Open   GOOG         8.983000e+01         9.101000e+01   
31            Open   META         1.228200e+02         1.273800e+02   
33            Open   TSLA         1.184700e+02         1.091100e+02   
34            Open    WMT         4.751667e+01         4.780667e+01   
35    Stock Splits   AMZN         0.000000e+00         0.000000e+00   
36    Stock Splits   GOOG         0.000000e+00         0.000000e+00   
37    Stock Splits   META         0.000000e+00         0.000000e+00   
38    Stock Splits   TSLA         0.000000e+00         0.000000e+00   
39    Stock Splits    WMT         0.000000e+00         0.000000e+00   
40          Volume   AMZN         7.670600e+07         6.888510e+07   
41          Volume   GOOG         2.073850e+07         2.704650e+07   
42          Volume   META         3.552850e+07         3.239710e+07   
44          Volume   TSLA         2.314028e+08         1.803890e+08   
45          Volume    WMT         1.724970e+07         1.333830e+07   

Date  2023-01-05 00:00:00  2023-01-06 00:00:00  2023-01-09 00:00:00  \
0            8.312000e+01         8.608000e+01         8.736000e+01   
1            8.636095e+01         8.774441e+01         8.838139e+01   
2            1.263445e+02         1.294101e+02         1.288627e+02   
4            1.103400e+02         1.130600e+02         1.197700e+02   
5            4.635476e+01         4.749041e+01         4.689831e+01   
6            8.312000e+01         8.608000e+01         8.736000e+01   
7            8.677000e+01         8.816000e+01         8.880000e+01   
8            1.269400e+02         1.300200e+02         1.294700e+02   
10           1.103400e+02         1.130600e+02         1.197700e+02   
11           4.775667e+01         4.892667e+01         4.831667e+01   
12           0.000000e+00         0.000000e+00         0.000000e+00   
13           0.000000e+00         0.000000e+00         0.000000e+00   
14           0.000000e+00         0.000000e+00         0.000000e+00   
15           0.000000e+00         0.000000e+00         0.000000e+00   
16           0.000000e+

### Final Dataframe for YFinance

In [ ]:
# Getting adj close and delta, volume, dividends, stock

# Will pull first 60 days
start_date = "2025-01-20"
end_date = "2025-03-22" # Need to specify March 22nd to pull previous day

companies = r3000_tickers
tickers = yf.Tickers(companies)
tickers_hist = tickers.history(period='max',start=start_date, end=end_date,interval='1d', auto_adjust=False)

adj_close = tickers_hist['Adj Close'].iloc[:]
adj_close = adj_close.T.reset_index().iloc[:,[0,1,43]]
adj_close.columns = ['symbol', 'start_adj_close', 'end_adj_close']
adj_close['delta_adj_close'] = adj_close['end_adj_close'] - adj_close['start_adj_close']
# adj_close

volume = tickers_hist['Volume'].iloc[:]
volume = volume.T.reset_index().iloc[:,[0,1,43]]
volume.columns = ['symbol', 'start_vol', 'end_vol']
volume['delta_vol'] = volume['end_vol'] - volume['start_vol']
# volume

dividends = tickers_hist['Dividends'].iloc[:]
dividends = dividends.T.reset_index()
div1 = dividends.iloc[:,0]
div2 = dividends.iloc[:,1:43].apply(np.count_nonzero, axis=1)
div = pd.concat([div1, div2], axis=1)
div.columns = ['symbol', 'div_count']
div['div_action'] = np.where(div['div_count'] > 0, 1, 0) 
# div

splits = tickers_hist['Stock Splits'].iloc[:]
splits = splits.T.reset_index()
sp1 = splits.iloc[:,0]
sp2 = splits.iloc[:,1:43].apply(np.count_nonzero, axis=1)
sp = pd.concat([sp1, sp2], axis=1)
sp.columns = ['symbol', 'split_count']
sp['split_action'] = np.where(sp['split_count'] > 0, 1, 0) 
# sp

adj_close

[*********************100%***********************]  2673 of 2673 completed

25 Failed downloads:
['BRKB', 'LGFB', 'MSFUT', 'METCV', 'LGFA', 'LENB', 'BFA', 'BFB', 'P5N994', 'GEFB', 'HEIA', 'CWENA', '-', 'RTYM5', 'GTXI', 'ADRO', 'XTSLA', 'ESM5', 'UHALB', 'MOGA']: YFTzMissingError('possibly delisted; no timezone found')
['INH', 'CBSH', 'METCB', 'CLSKW']: YFPricesMissingError('possibly delisted; no price data found  (1d 2025-01-20 -> 2025-03-22)')
['\xa0']: YFTzMissingError('$\xa0: possibly delisted; no timezone found')


symbol  start_adj_close  end_adj_close  delta_adj_close
0         -              NaN            NaN              NaN
1         A       152.246552     120.494011       -31.752541
2        AA        38.486763      33.950001        -4.536762
3       AAL        18.639999      11.390000        -7.249999
4      AAMI        25.589302      25.530001        -0.059301
...     ...              ...            ...              ...
2668   ZVRA         8.150000       8.010000        -0.139999
2669    ZWS        38.741974      33.459999        -5.281975
2670   ZYME        14.290000      12.720000        -1.570000
2671   ZYXI         8.140000       2.450000        -5.690000
2672                     NaN            NaN              NaN

[2673 rows x 4 columns]

### It will take about 1 hour to completely run through all market caps

In [ ]:
# Retrieving Market Cap from YFinance
def get_market_caps(tickers):
    market_caps = {}
    for ticker in tickers:
        try:
            stock = yf.Ticker(ticker)
            info = stock.info
            market_cap = info.get("marketCap")
            if market_cap is not None:
                market_caps[ticker] = market_cap
            else:
                market_caps[ticker] = None
        except Exception as e:
            print(f"Error getting data for {ticker}: {e}")
            market_caps[ticker] = None
        time.sleep(1)
    df = pd.DataFrame(list(market_caps.items()), columns=['Ticker', 'Market Cap'])
    return df

In [ ]:
stock_tickers = r3000_tickers
market_cap_df = get_market_caps(stock_tickers)

if market_cap_df is not None:
    print(market_cap_df)

404 Client Error: Not Found for url: https://query2.finance.yahoo.com/v10/finance/quoteSummary/XTSLA?modules=financialData%2CquoteType%2CdefaultKeyStatistics%2CassetProfile%2CsummaryDetail&corsDomain=finance.yahoo.com&formatted=false&symbol=XTSLA&crumb=Rf9oudV5pAv


Error getting data for XTSLA: 'NoneType' object has no attribute 'update'


401 Client Error: Unauthorized for url: https://query1.finance.yahoo.com/v7/finance/quote?symbols=HAS&formatted=false&crumb=Too+Many+Requests%0D%0A


Error getting data for HAS: 'NoneType' object is not iterable


401 Client Error: Unauthorized for url: https://query2.finance.yahoo.com/v10/finance/quoteSummary/MOS?modules=financialData%2CquoteType%2CdefaultKeyStatistics%2CassetProfile%2CsummaryDetail&corsDomain=finance.yahoo.com&formatted=false&symbol=MOS&crumb=Too+Many+Requests%0D%0A


Error getting data for MOS: 'NoneType' object has no attribute 'update'


401 Client Error: Unauthorized for url: https://query2.finance.yahoo.com/v10/finance/quoteSummary/HSIC?modules=financialData%2CquoteType%2CdefaultKeyStatistics%2CassetProfile%2CsummaryDetail&corsDomain=finance.yahoo.com&formatted=false&symbol=HSIC&crumb=Too+Many+Requests%0D%0A


Error getting data for HSIC: 'NoneType' object has no attribute 'update'


404 Client Error: Not Found for url: https://query2.finance.yahoo.com/v10/finance/quoteSummary/MSFUT?modules=financialData%2CquoteType%2CdefaultKeyStatistics%2CassetProfile%2CsummaryDetail&corsDomain=finance.yahoo.com&formatted=false&symbol=MSFUT&crumb=Rf9oudV5pAv


Error getting data for MSFUT: 'NoneType' object has no attribute 'update'


404 Client Error: Not Found for url: https://query2.finance.yahoo.com/v10/finance/quoteSummary/LENB?modules=financialData%2CquoteType%2CdefaultKeyStatistics%2CassetProfile%2CsummaryDetail&corsDomain=finance.yahoo.com&formatted=false&symbol=LENB&crumb=Rf9oudV5pAv


Error getting data for LENB: 'NoneType' object has no attribute 'update'


404 Client Error: Not Found for url: https://query2.finance.yahoo.com/v10/finance/quoteSummary/GEFB?modules=financialData%2CquoteType%2CdefaultKeyStatistics%2CassetProfile%2CsummaryDetail&corsDomain=finance.yahoo.com&formatted=false&symbol=GEFB&crumb=Rf9oudV5pAv


Error getting data for GEFB: 'NoneType' object has no attribute 'update'


404 Client Error: Not Found for url: https://query2.finance.yahoo.com/v10/finance/quoteSummary/METCV?modules=financialData%2CquoteType%2CdefaultKeyStatistics%2CassetProfile%2CsummaryDetail&corsDomain=finance.yahoo.com&formatted=false&symbol=METCV&crumb=Rf9oudV5pAv


Error getting data for METCV: 'NoneType' object has no attribute 'update'


404 Client Error: Not Found for url: https://query2.finance.yahoo.com/v10/finance/quoteSummary/P5N994?modules=financialData%2CquoteType%2CdefaultKeyStatistics%2CassetProfile%2CsummaryDetail&corsDomain=finance.yahoo.com&formatted=false&symbol=P5N994&crumb=Rf9oudV5pAv


Error getting data for P5N994: 'NoneType' object has no attribute 'update'


404 Client Error: Not Found for url: https://query2.finance.yahoo.com/v10/finance/quoteSummary/RTYM5?modules=financialData%2CquoteType%2CdefaultKeyStatistics%2CassetProfile%2CsummaryDetail&corsDomain=finance.yahoo.com&formatted=false&symbol=RTYM5&crumb=Rf9oudV5pAv


Error getting data for RTYM5: 'NoneType' object has no attribute 'update'


404 Client Error: Not Found for url: https://query2.finance.yahoo.com/v10/finance/quoteSummary/ESM5?modules=financialData%2CquoteType%2CdefaultKeyStatistics%2CassetProfile%2CsummaryDetail&corsDomain=finance.yahoo.com&formatted=false&symbol=ESM5&crumb=Rf9oudV5pAv


Error getting data for ESM5: 'NoneType' object has no attribute 'update'


404 Client Error: Not Found for url: https://query2.finance.yahoo.com/v10/finance/quoteSummary/%C2%A0?modules=financialData%2CquoteType%2CdefaultKeyStatistics%2CassetProfile%2CsummaryDetail&corsDomain=finance.yahoo.com&formatted=false&symbol=%C2%A0&crumb=Rf9oudV5pAv


Error getting data for  : 'NoneType' object has no attribute 'update'
     Ticker    Market Cap
0      AAPL  2.829863e+12
1      MSFT  2.675043e+12
2      NVDA  2.301164e+12
3      AMZN  1.812207e+12
4      META  1.278814e+12
...     ...           ...
2668   GTXI           NaN
2669  WLLBW           NaN
2670  RTYM5           NaN
2671   ESM5           NaN
2672                  NaN

[2673 rows x 2 columns]


### Classify the Cap Size based on Market Cap (previous cell)

In [ ]:
def classify_cap_size(adj_market_cap):
    if adj_market_cap >= 10_000_000_000:
        return 'Large Cap'
    elif 2_000_000_000 <= adj_market_cap < 10_000_000_000:
        return 'Medium Cap'
    elif adj_market_cap < 2_000_000_000:
        return 'Small Cap'
    else:
        return None

In [ ]:
market_cap_df = pd.read_csv('r3000_market_cap.csv')
market_cap_df.columns = ['symbol','market_cap']
market_cap_df['cap_size'] = market_cap_df['market_cap'].apply(classify_cap_size)
market_cap_df

symbol    market_cap   cap_size
0      AAPL  2.829863e+12  Large Cap
1      MSFT  2.675043e+12  Large Cap
2      NVDA  2.301164e+12  Large Cap
3      AMZN  1.812207e+12  Large Cap
4      META  1.278814e+12  Large Cap
...     ...           ...        ...
2668   GTXI           NaN       None
2669  WLLBW           NaN       None
2670  RTYM5           NaN       None
2671   ESM5           NaN       None
2672                  NaN       None

[2673 rows x 3 columns]

### Final Dataset

In [ ]:
# Merge all datasets

merged_df = pd.merge(r3000_df, emp_df, on='symbol')
merged_df = pd.merge(merged_df, adj_close, on='symbol')
merged_df = pd.merge(merged_df, volume, on='symbol')
merged_df = pd.merge(merged_df, div, on='symbol')
merged_df = pd.merge(merged_df, sp, on='symbol')
merged_df = pd.merge(merged_df, market_cap_df, on='symbol')
merged_df

symbol                        name                  sector  \
0      AAPL                   APPLE INC  Information Technology   
1      MSFT              MICROSOFT CORP  Information Technology   
2      NVDA                 NVIDIA CORP  Information Technology   
3      AMZN              AMAZON COM INC  Consumer Discretionary   
4      META  META PLATFORMS INC CLASS A           Communication   
...     ...                         ...                     ...   
2550   BLUE            BLUEBIRD BIO INC             Health Care   
2551   MODV               MODIVCARE INC             Health Care   
2552   QTTB                 Q32 BIO INC             Health Care   
2553   AGEN                  AGENUS INC             Health Care   
2554    PBI            PITNEY BOWES INC             Industrials   

                          exchange  emp_count  start_adj_close  end_adj_close  \
0                           NASDAQ     150000       222.395477     218.270004   
1                           NASDAQ     228000       427.642517     391.260010   
2                           NASDAQ      29600       140.817047     117.699997   
3                           NASDAQ    1551000       230.710007     196.210007   
4                           NASDAQ      72404       615.912048     596.250000   
...                            ...        ...              ...            ...   
2550                        NASDAQ        375         8.140000       3.900000   
2551                        NASDAQ      21200         5.940000       2.120000   
2552                        NASDAQ         41         3.400000       2.070000   
2553                        NASDAQ        389         3.830000       1.610000   
2554  New York Stock Exchange Inc.      10500         7.388297       9.380000   

      delta_adj_close    start_vol      end_vol   delta_vol  div_count  \
0           -4.125473   98070400.0   94127800.0  -3942600.0          1   
1          -36.382507   26085700.0   39675900.0  13590200.0          1   
2          -23.117050  197749000.0  266498500.0  68749500.0          1   
3          -34.500000   39951500.0   60056900.0  20105400.0          0   
4          -19.662048   11674000.0   25015900.0  13341900.0          1   
...               ...          ...          ...         ...        ...   
2550        -4.240000     112300.0     153800.0     41500.0          0   
2551        -3.820000     704200.0     492900.0   -211300.0          0   
2552        -1.330000     189900.0      75500.0   -114400.0          0   
2553        -2.220000     808200.0     733900.0    -74300.0          0   
2554         1.991703    2232500.0    4152400.0   1919900.0          1   

      div_action  split_count  split_action    market_cap   cap_size  
0              1            0             0  2.829863e+12  Large Cap  
1              1            0             0  2.675043e+12  Large Cap  
2              1            0             0  2.301164e+12  Large Cap  
3              0            0             0  1.812207e+12  Large Cap  
4              1            0             0  1.278814e+12  Large Cap  
...          ...          ...           ...           ...        ...  
2550           0            0             0  4.846084e+07  Small Cap  
2551           0            0             0  1.613250e+07  Small Cap  
2552           0            0             0  2.000406e+07  Small Cap  
2553           0            0             0  3.948173e+07  Small Cap  
2554           1            0             0  1.456919e+09  Small Cap  

[2555 rows x 17 columns]

### Merged DF stats
3 stocks that could not have their information downloaded
118 that have missing information and therefore dropped during the merge

Leaving 2555 observations


In [ ]:
# Compare Blackrock stocks to Final Merged DF
unique_df1 = set(r3000_df['symbol'].unique())
unique_df2 = set(merged_df['symbol'].unique())

print("Distinct values in df1['symbol']:", unique_df1)
print("Distinct values in df2['symbol']:", unique_df2)

# Find the differences
in_df1_not_df2 = unique_df1 - unique_df2
in_df2_not_df1 = unique_df2 - unique_df1
common_values = unique_df1.intersection(unique_df2)

print("Values in df1 but not df2", in_df1_not_df2)
print("Values in df2 but not df1", in_df2_not_df1)
print("Common values:", common_values)